icicle/appUtils/large_ntt/verification.cu

```diff
@@ -0,0 +1,160 @@
+
+#define CURVE_ID BLS12_381
+
+#include "../../primitives/field.cuh"
+#include "../../primitives/projective.cuh"
+#include "../../utils/cuda_utils.cuh"
+#include <chrono>
+#include <iostream>
+#include <vector>
+
+#include "curves/curve_config.cuh"
+#include "ntt/ntt.cu"
+#include "ntt/ntt_impl.cuh"
+#include <memory>
+
+#define PERFORMANCE
+
+typedef curve_config::scalar_t test_scalar;
+typedef curve_config::scalar_t test_data; // uncomment for NTT
+// typedef curve_config::projective_t test_data; // uncomment for ECNTT
+#include "kernel_ntt.cu"
+
+#define $CUDA(call)                                                                                                    \
+  if ((call) != 0) {                                                                                                   \
+    printf("\nCall \"" #call "\" failed from %s, line %d, error=%d\n", __FILE__, __LINE__, cudaGetLastError());        \
+    exit(1);                                                                                                           \
+  }
+
+void random_samples(test_data* res, uint32_t count)
+{
+  for (int i = 0; i < count; i++)
+    res[i] = i < 1000 ? test_data::rand_host() : res[i - 1000];
+}
+
+void incremental_values(test_scalar* res, uint32_t count)
+{
+  for (int i = 0; i < count; i++) {
+    res[i] = i ? res[i - 1] + test_scalar::one() * test_scalar::omega(4) : test_scalar::zero();
+  }
+}
+
+int main(int argc, char** argv)
+{
+#ifdef PERFORMANCE
+  cudaEvent_t icicle_start, icicle_stop, new_start, new_stop;
+  float icicle_time, new_time;
+#endif
+
+  int NTT_LOG_SIZE = (argc > 1) ? atoi(argv[1]) : 19; // assuming second input is the log-size
+  int NTT_SIZE = 1 << NTT_LOG_SIZE;
+  bool INPLACE = (argc > 2) ? atoi(argv[2]) : true;
+  int INV = (argc > 3) ? atoi(argv[3]) : true;
```

Comment:
the verification is missing the DIF case and the NR,RN cases (that also checks the normalization kernel)

Reply:
right. This will be another PR as we agreed in the team.